packages/amplify-category-auth/src/provider-utils/awscloudformation/utils/synthesize-resources.ts

```diff
@@ -0,0 +1,340 @@
+      target: `${targetDir}/function/${functionName}/src/package.json`,
+    },
+    {
+      dir: adminAuthAssetRoot,
+      template: 'admin-queries-function-template.json.ejs',
+      target: `${targetDir}/function/${functionName}/${functionName}-cloudformation-template.json`,
+    },
+  ];
+
+  // copy over the files
+  await context.amplify.copyBatch(context, copyJobs, functionProps, true);
+
+  if (operation === 'add') {
+    // add amplify-meta and backend-config
+    const backendConfigs = {
+      service: FunctionServiceName.LambdaFunction,
+      providerPlugin: 'awscloudformation',
+      build: true,
+      dependsOn,
+    };
+
+    await context.amplify.updateamplifyMetaAfterResourceAdd('function', functionName, backendConfigs);
+    context.print.success(`Successfully added ${functionName} function locally`);
+  } else {
+    context.print.success(`Successfully updated ${functionName} function locally`);
+  }
+};
+
+export const createAdminAuthAPI = async (context: any, authResourceName: string, functionName: string, operation: 'update' | 'add') => {
+  const targetDir = context.amplify.pathManager.getBackendDirPath();
+  const pluginDir = __dirname;
+  const apiName = 'AdminQueries';
+  const dependsOn = [];
+
+  dependsOn.push(
+    {
+      category: 'auth',
+      resourceName: authResourceName,
+      attributes: ['UserPoolId'],
+    },
+    {
+      category: 'function',
+      resourceName: functionName,
+      attributes: ['Arn', 'Name'],
+    },
+  );
+
+  const apiProps = {
+    functionName,
+    authResourceName,
+    dependsOn,
+  };
+
+  const copyJobs = [
+    {
+      dir: adminAuthAssetRoot,
+      template: 'admin-queries-api-template.json.ejs',
+      target: `${targetDir}/api/${apiName}/admin-queries-cloudformation-template.json`,
+    },
+    {
+      dir: adminAuthAssetRoot,
+      template: 'admin-queries-api-params.json',
+      target: `${targetDir}/api/${apiName}/parameters.json`,
+    },
+  ];
+
+  // copy over the files
+  await context.amplify.copyBatch(context, copyJobs, apiProps, true);
+
+  if (operation === 'add') {
+    // Update amplify-meta and backend-config
+    const backendConfigs = {
+      service: 'API Gateway',
+      providerPlugin: 'awscloudformation',
+      dependsOn,
+    };
+
+    await context.amplify.updateamplifyMetaAfterResourceAdd('api', apiName, backendConfigs);
+    context.print.success(`Successfully added ${apiName} API locally`);
+  } else {
+    context.print.success(`Successfully updated ${apiName} API locally`);
+  }
+};
+
+export const copyCfnTemplate = async (context: any, category: string, options: any, cfnFilename: string) => {
+  const { amplify } = context;
+  const targetDir = amplify.pathManager.getBackendDirPath();
+
+  const copyJobs = [
+    {
+      dir: cfnTemplateRoot,
+      template: cfnFilename,
+      target: `${targetDir}/${category}/${options.resourceName}/${options.resourceName}-cloudformation-template.yml`,
```

Comment:
`path.join` ?

Reply:
Copied from existing code, but I can update it